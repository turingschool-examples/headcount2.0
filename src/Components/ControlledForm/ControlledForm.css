input {
  border-radius: 18px;
  color: #BBBDBF;
  display: block;
  font-size: 16px;
  font-weight: 300;
  margin: 50px auto;
  padding: 5px;
  width: 50%;
}
