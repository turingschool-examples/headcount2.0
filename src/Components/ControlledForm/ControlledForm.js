import React, { Component } from 'react';
import './ControlledForm.css';
import PropTypes from 'prop-types';

class ControlledForm extends Component {

  handleChange = (event) => {
    const input = event.target.value;
    this.props.findDistrict(input)
  }

  render() {
    return (
      <form>
        <input 
          onChange={this.handleChange} 
          placeholder='Enter District'
        />
      </form>
    )
  }
}

  ControlledForm.propTypes = {
  findDistrict: PropTypes.func.isRequired
}

export default ControlledForm