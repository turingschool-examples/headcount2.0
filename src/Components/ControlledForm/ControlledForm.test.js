import React from 'react';
import { shallow, mount } from 'enzyme';
import ControlledForm from './ControlledForm'

describe('ControlledForm', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<ControlledForm />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should call findDistrict prop function with the proper arguments', () => {
    const mockChange = jest.fn();
    const wrapper = mount(<ControlledForm findDistrict={mockChange}/>);

    const mockEvent = { target: {value:'COLORADO'}};
   
    wrapper.instance().handleChange(mockEvent);
    expect(mockChange).toBeCalledWith('COLORADO');

  })
})