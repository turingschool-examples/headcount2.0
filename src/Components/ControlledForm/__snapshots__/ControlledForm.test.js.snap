// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlledForm should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ControlledForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        onChange={[Function]}
        placeholder="Enter District"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "onChange": [Function],
        "placeholder": "Enter District",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          onChange={[Function]}
          placeholder="Enter District"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Enter District",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
